odl/tomo/geometry/parallel.py

```diff
@@ -1468,6 +1470,213 @@ def __getitem__(self, indices):
+        >>> dir = geom.det_to_src(0.5, 0)  # interpolate
+        >>> np.allclose(dir, [-1, 1] / np.linalg.norm([-1, 1]))
+        True
+
+        Both variables support vectorized calls, i.e., stacks of
+        parameters can be provided. The order of axes in the output (left
+        of the ``ndim`` axis for the vector dimension) corresponds to the
+        order of arguments:
+        TODO: adapt this
+
+        >>> dirs = geom.det_to_src(0, [-1, 0, 0.5, 1])
+        >>> dirs
+        array([[ 0., -1.],
+               [ 0., -1.],
+               [ 0., -1.],
+               [ 0., -1.]])
+        >>> dirs.shape  # (num_dparams, ndim)
+        (4, 2)
+        >>> dirs = geom.det_to_src([0, np.pi / 2, np.pi], 0)
+        >>> np.allclose(dirs, [[0, -1],
+        ...                    [1, 0],
+        ...                    [0, 1]])
+        True
+        >>> dirs.shape  # (num_angles, ndim)
+        (3, 2)
+        >>> # Providing 3 pairs of parameters, resulting in 3 vectors
+        >>> dirs = geom.det_to_src([0, np.pi / 2, np.pi], [-1, 0, 1])
+        >>> dirs[0]  # Corresponds to angle = 0, dparam = -1
+        array([ 0., -1.])
+        >>> dirs.shape
+        (3, 2)
+        >>> # Pairs of parameters arranged in arrays of same size
+        >>> geom.det_to_src(np.zeros((4, 5)), np.zeros((4, 5))).shape
+        (4, 5, 2)
+        >>> # "Outer product" type evaluation using broadcasting
+        >>> geom.det_to_src(np.zeros((4, 1)), np.zeros((1, 5))).shape
+        (4, 5, 2)
+        """
+        squeeze_index = (np.shape(index) == ())
+        index_in = index
+        index = np.array(index, dtype=float, copy=False, ndmin=1)
+
+        dparam_in = dparam
+        if self.det_params.ndim == 1:
+            squeeze_dparam = (np.shape(dparam) == ())
+            dparam = np.array(dparam, dtype=float, copy=False, ndmin=1)
+        else:
+            squeeze_dparam = (np.broadcast(*dparam).shape == ())
+            dparam = tuple(np.array(p, dtype=float, copy=False, ndmin=1)
+                           for p in dparam)
+
+        if self.check_bounds:
+            if not is_inside_bounds(index, self.motion_params):
+                raise ValueError('`index` {} not in the valid range {}'
+                                 ''.format(index_in, self.motion_params))
+
+            if not is_inside_bounds(dparam, self.det_params):
+                raise ValueError('`dparam` {} not in the valid range {}'
+                                 ''.format(dparam_in, self.det_params))
+
+        at_max_flat = (index == self.motion_params.max_pt).ravel()
+
+        if self.det_params.ndim == 1:
+            bcast_shape = np.broadcast(index, dparam).shape
+        else:
+            bcast_shape = np.broadcast(index, *dparam).shape
+
+        # TODO: shapes are not quite right here. Scalar works, but
+        # vectorized goes wrong
+        ray_dir = np.empty(bcast_shape + (self.ndim,), dtype=float)
+        print('result shape:', ray_dir.shape)
```

Comment:
Debug statement, TODO remove

Regarding vectorization, I'd suggest putting it in a separate PR.